flink-python/src/main/java/org/apache/flink/table/runtime/typeutils/PythonTypeUtils.java

```diff
@@ -0,0 +1,473 @@
+import org.apache.flink.api.common.typeutils.base.array.BytePrimitiveArraySerializer;
+import org.apache.flink.api.java.typeutils.runtime.RowSerializer;
+import org.apache.flink.fnexecution.v1.FlinkFnApi;
+import org.apache.flink.table.runtime.typeutils.serializers.BaseRowSerializer;
+import org.apache.flink.table.runtime.typeutils.serializers.DateSerializer;
+import org.apache.flink.table.runtime.typeutils.serializers.StringSerializer;
+import org.apache.flink.table.types.logical.AnyType;
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.BigIntType;
+import org.apache.flink.table.types.logical.BinaryType;
+import org.apache.flink.table.types.logical.BooleanType;
+import org.apache.flink.table.types.logical.CharType;
+import org.apache.flink.table.types.logical.DateType;
+import org.apache.flink.table.types.logical.DayTimeIntervalType;
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.DistinctType;
+import org.apache.flink.table.types.logical.DoubleType;
+import org.apache.flink.table.types.logical.FloatType;
+import org.apache.flink.table.types.logical.IntType;
+import org.apache.flink.table.types.logical.LocalZonedTimestampType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeVisitor;
+import org.apache.flink.table.types.logical.MapType;
+import org.apache.flink.table.types.logical.MultisetType;
+import org.apache.flink.table.types.logical.NullType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.logical.SmallIntType;
+import org.apache.flink.table.types.logical.StructuredType;
+import org.apache.flink.table.types.logical.SymbolType;
+import org.apache.flink.table.types.logical.TimeType;
+import org.apache.flink.table.types.logical.TimestampType;
+import org.apache.flink.table.types.logical.TinyIntType;
+import org.apache.flink.table.types.logical.VarBinaryType;
+import org.apache.flink.table.types.logical.VarCharType;
+import org.apache.flink.table.types.logical.YearMonthIntervalType;
+import org.apache.flink.table.types.logical.ZonedTimestampType;
+
+/**
+ * Utilities for serializing Flink data.
+ */
+@Internal
+public final class PythonTypeUtils {
+
+	private static final String EMPTY_STRING = "";
+
+	public static TypeSerializer toFlinkTypeSerializer(LogicalType logicalType) {
+		return logicalType.accept(new LogicalTypeToTypeSerializerConverter());
+	}
+
+	public static TypeSerializer toBlinkTypeSerializer(LogicalType logicalType) {
+		return logicalType.accept(new LogicalTypeToBlinkTypeSerializerConverter());
+	}
+
+	public static FlinkFnApi.Schema.FieldType toProtoType(LogicalType logicalType) {
+		return logicalType.accept(new LogicalTypeToProtoTypeConverter());
+	}
+
+	private static class LogicalTypeToTypeSerializerConverter implements LogicalTypeVisitor<TypeSerializer> {
```

Comment:
Maybe extend from `LogicalTypeDefaultVisitor` is better, so that we don't need to override the type that we have not supported explicitly and throw the exception in `defaultMethod`.